modules/populateItems.js

```diff
@@ -3,62 +3,49 @@ import {
   hideChecked,
   sectionSort,
   itemsDiv,
-  priceAmount,
+  stickyQuickSortFooter,
+  quickSortDiv,
 } from './domElements.js';
 import db from './db.js';
 
-// fetch items from the database
-export const populateItems = async () => {
-  const allItems = await db.items.reverse().toArray();
-  let sortedItems = allItems.map((item) => ({
-    ...item,
-    // add false - as this doesn't come back at all if not checked and set
-    isPurchased: item.isPurchased ?? false,
-  }));
+// TODO: Modularize a bunch of stuff here - footer code, sorting functions, etc
 
-  if (hideChecked.checked) {
-    sortedItems = sortedItems.filter((item) => !item.isPurchased);
-  }
+export let selectedSection = null;
 
-  // if both auto sort and section sort are turned on
-  if (autoSort.checked && sectionSort.checked) {
-    // create lists of the checked and unchecked items
-    const uncheckedItems = sortedItems.filter((item) => !item.isPurchased);
-    const checkedItems = sortedItems.filter((item) => item.isPurchased);
+export const populateItems = async () => {
+  // fetch items from the database
+  const allItems = await db.items.reverse().toArray();
+  let sortedItems;
+  // filter immediately on hide checked and section
+  const filteredItems = filterItems(allItems, hideChecked, selectedSection);
```

Comment:
There's an issue here - if we select a section from the footer, and then clear out all the items from that section, we see a blank list. 

The expected behavior should probably be to clear the section selection and render all items again.